docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,264 @@
+We assume that the destination process implicitly verifies authenticity upon calling `Deliver` for any datagram.
+
+The communication layer is _asynchronous_.
+There is no guarantee on how much time it will take for the destination of a datagram to deliver that datagram.
+
+
+###### Fault assumptions
+This specification operates under the assumption that processes `I` and `R` are not faulty, i.e., do not deviate from any protocol we describe below.
+Beside these two processes, the system comprises an arbitrary number of other chains, any of which may be Byzantine; these other chains may, for instance, try to impersonate `I` or `R`.
+
+
+###### Additional remarks on the system model
+We explicitly overlook here two assumptions of synchrony that may be necessary for achieving liveness of **other** protocols, but they are not necessary within the connection handshake protocol.
+These assumptions are: (i) partial synchrony is necessary for achieving liveness of a chain if that chain is implemented using a deterministic consensus protocol (e.g., Tendermint Core); (ii) synchrony is necessary for guaranteeing message delivery in any connection-oriented protocol such as TCP, which is most likely the underlying communication layer running between `I` and `R`.
+
+
+#### Properties
+
+###### Preliminary definitions
+We introduce two abstractions that are relevant at this point:
+
+1. A `ConnectionEnd` is a data type (i.e., interface) that abstracts over the details of a connection at a process.
+One of the goals of the connection handshake protocol is to  allocate an object of this type at each of the initiator and responder process.
+
+2. A `State` is a data type that abstracts over the details of the state of a process at a current point in time (e.g., a summary or snapshot).
+In the context of the CH problem, a process uses this data type to (i) declare their state to another process, and (ii) represent the state of another process.
+
+Knowing the precise declaration of `ConnectionEnd` and `State` is not necessary in understanding the properties of the CH protocol.
+Suffice to say that each `ConnectionEnd` has associated the `State` of both the initiator and responder process.
+We will declare this data types later (see [2.2 Protocol](#22-protocol)).
+
+We now restate the connection handshake problem in a slightly more precise (or lower level) formulation:
+A connection handshake enables two processes to agree on a consistent view of their state, to verify each other's state, and to allocate a new connection.
+
+
+###### Guarantees
+We refine the safety guarantee **[CH/Seq/1/Agreement]**  (defined in the [sequential problem statement](#sequential-problem-statement)) by splitting this into four complementary safety properties:
+
+- **[CH/Proto/1/Uniqueness]**
+If a process finishes a handshake for a `ConnectionEnd` `c`, no handshake has previously finished for `c`.
```

Comment:
Would it be enough to say that `connectionEnd c` should be unique? Even intermediate stages of the handshake require there is no other handshake in progress or finished for `c`